testdata/scripts/make_vds.py

```diff
@@ -50,59 +68,66 @@ def __init__(self, vds_origin, inline_spacing, xline_spacing):
             self.xline_spacing  = xline_spacing
 
         @classmethod
-        # unfixed version copied right from previous PR
         def from_segy(cls, segy: segyio.SegyFile):
             cdp_x = segyio.su.cdpx
             cdp_y = segyio.su.cdpy
 
             origin_header = segy.header[0]
-            scale = abs(origin_header[segyio.su.scalco])
-
-            # casting genrator to list causing problem so we iterating using next fuction
-            il = segy.header.iline[segy.ilines[0]]
-            il_header = next(il)
-            il_header = next(il)
-
-            xl = segy.header.xline[segy.xlines[0]]
-            xl_dif = np.diff(segy.xlines).tolist()
-
-            # Calculating xline spacing the distance between the point in xlines direction
-            xline_spacing = (
-                ((il_header[cdp_x] / scale) - (origin_header[cdp_x] / scale))
-                / xl_dif[0],
-                ((il_header[cdp_y] / scale) - (origin_header[cdp_y] / scale))
-                / xl_dif[0],
+            scaler = abs(origin_header[segyio.su.scalco])
+            # As per the SEGY specifications, if the scaler value is negative,
+            # it should be divided, whereas if the value is positive, it should
+            # be multiplied. Since we only use negative scaler values, our
+            # computation logic only involves division. However, if we encounter
+            # positive scaler values, the computation logic will need to be
+            # adjusted to apply multiplication instead.
+            # ref https://library.seg.org/pb-assets/technical-standards/seg_y_rev1-1686080991247.pdf
+            # (page 14, line 71-72)
+
+            # Casting the generator into a list causes problems so we iterate using next function
+            ilines = segy.header.iline[segy.ilines[0]]
+            ilheader = next(ilines)
+            ilheader = next(ilines)
+
+            xlines = segy.header.xline[segy.xlines[0]]
+            xldiff = np.diff(segy.xlines).tolist()
```

Comment:
Note that this thing will calculate the difference between all elements in the array, which we do not need.

See if it makes sense to just do
```
ildiff = segy.ilines[1] - segy.ilines[0]
xldiff = segy.xlines[1] - segy.xlines[0]
```

to get one value